saw-core/src/Verifier/SAW/Term/Functor.hs

```diff
@@ -455,14 +461,9 @@ instance Hashable Term where
   -- hashes introduces less randomness/freshness, it maintains plenty, and
   -- provides benefits as described above. No code should ever rely on total
   -- uniqueness of hashes, and terms are no exception.
-  hashWithSalt salt STApp{ stAppHash = h } = salt `combine` 0x00000000 `hashWithSalt` h
-  hashWithSalt salt (Unshared t) = salt `combine` 0x55555555 `hashWithSalt` hash t
-
-
--- | Combine two given hash values.  'combine' has zero as a left
--- identity. (FNV hash, copied from Data.Hashable 1.2.1.0.)
-combine :: Int -> Int -> Int
-combine h1 h2 = (h1 * 0x01000193) `xor` h2
+  hash STApp{ stAppHash = h } = h
+  hash (Unshared t) = hash t
+  hashWithSalt salt = hashWithSalt salt . hash
```

Comment:
Similarly, leave some commentary about why this is defined like it is, and reference `alphaEquiv`.